UnitySDK/Assets/ML-Agents/Examples/SharedAssets/Scripts/ModelOverrider.cs

```diff
@@ -0,0 +1,110 @@
+    {
+        const string k_CommandLineFlag = "--mlagents-override-model";
+        // Assets paths to use, with the behavior name as the key.
+        Dictionary<string, string> m_BehaviorNameOverrides = new Dictionary<string, string>();
+
+        // Cached loaded NNModels, with the behavior name as the key.
+        Dictionary<string, NNModel> m_CachedModels = new Dictionary<string, NNModel>();
+
+        /// <summary>
+        /// Get the asset path to use from the commandline arguments.
+        /// </summary>
+        /// <returns></returns>
+        void GetAssetPathFromCommandLine()
+        {
+            m_BehaviorNameOverrides.Clear();
+            m_BehaviorNameOverrides["3DBall"] = "/Users/chris.elion/code/ml-agents/models/ppo/3DBall.nn"; // TODO REMOVE ME
+
+            var args = Environment.GetCommandLineArgs();
+            for (var i = 0; i < args.Length-2; i++)
+            {
+                if (args[i] == k_CommandLineFlag)
+                {
+                    var key = args[i + 1].Trim();
+                    var value = args[i + 2].Trim();
+                    m_BehaviorNameOverrides[key] = value;
+                }
+            }
+        }
+
+        void OnEnable()
+        {
+            GetAssetPathFromCommandLine();
+            if (m_BehaviorNameOverrides.Count > 0)
+            {
+                OverrideModel();
+            }
+        }
+
+        NNModel GetModelForBehaviorName(string behaviorName)
+        {
+            if (m_CachedModels.ContainsKey(behaviorName))
+            {
+                return m_CachedModels[behaviorName];
+            }
+
+            if (!m_BehaviorNameOverrides.ContainsKey(behaviorName))
+            {
+                Debug.Log($"No override for behaviorName {behaviorName}");
+                return null;
+            }
+
+            var assetPath = m_BehaviorNameOverrides[behaviorName];
+
+            byte[] model = null;
+            try
+            {
+                model = File.ReadAllBytes(assetPath);
+            }
+            catch(IOException e)
+            {
+                Debug.Log($"Couldn't load file {assetPath}", this);
+                // Cache the null so we don't repeatedly try to load a missing file
+                m_CachedModels[behaviorName] = null;
+                return null;
+            }
+
+            var asset = ScriptableObject.CreateInstance<NNModel>();
+            asset.Value = model;
+            asset.name = "Override - " + Path.GetFileName(assetPath);
+            m_CachedModels[behaviorName] = asset;
+            return asset;
+        }
+
+        /// <summary>
+        /// Load the NNModel file from the specified path, and give it to the attached agent.
+        /// </summary>
+        void OverrideModel()
+        {
+            var agent = GetComponent<Agent>();
+            agent.LazyInitialize();
+            var bp = agent.GetComponent<BehaviorParameters>();
+
+            var behaviorNameAndTeamId = bp.behaviorName;
```

Comment:
Going to revisit this later, but I think we need some better accessors on BehaviorParameters.